cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/TimePartitionedFileSetDatasetAvroSink.java

```diff
@@ -40,43 +43,64 @@
 
-  private static final String TPFS_NAME = "name";
-  private static final String SCHEMA = "schema";
-  private static final String BASE_PATH = "basePath";
+  private static final String SCHEMA_DESC = "The schema of the record";
+  private static final String TPFS_NAME_DESC = "Name of the Time Partitioned FileSet Dataset to which the records " +
+    "have to be written";
+  private static final String BASE_PATH_DESC = "Optional: The base path for the time partitioned fileset. Should " +
+    "be provided if you want the pipeline to create the fileset.";
 
-  @Override
-  public void configure(StageConfigurer configurer) {
-    configurer.setName(TimePartitionedFileSetDatasetAvroSink.class.getSimpleName());
-    configurer.setDescription("AVRO Sink with Time Partitioned File Dataset");
-    configurer.addProperty(new Property(SCHEMA, "The schema of the record", true));
-    configurer.addProperty(new Property(TPFS_NAME, "Name of the Time Partitioned FileSet Dataset to which the " +
-      "records have to be written", true));
-    configurer.addProperty(new Property(BASE_PATH, "Optional: The base path for the time partitioned fileset. Should" +
-      "be provided if you want the pipeline to create the fileset.", false));
+  /**
+   * Config for TimePartitionedFileSetDatasetAvroSink
+   */
+  public static class TimePartitionedFileSetDatasetAvroConfig extends PluginConfig {
+
+    @Name(Properties.TimePartitionedFileSetDataset.TPFS_NAME)
+    @Description(TPFS_NAME_DESC)
+    private String tpfsName;
+
+    @Name(Properties.TimePartitionedFileSetDataset.SCHEMA)
+    @Description(SCHEMA_DESC)
+    private String schema;
+
+    @Name(Properties.TimePartitionedFileSetDataset.BASE_PATH)
+    @Description(BASE_PATH_DESC)
+    @Nullable
+    private String basePath;
+
+    public TimePartitionedFileSetDatasetAvroConfig(String tpfsName, String basePath, String schema) {
+      this.tpfsName = tpfsName;
+      this.basePath = basePath;
+      this.schema = schema;
+    }
   }
 
   @Override
   public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
     // if the base path is provided then we should try to create the fileset here
-    if (!Strings.isNullOrEmpty(stageConfig.getProperties().get(BASE_PATH))) {
-      String tpfsName = stageConfig.getProperties().get(TPFS_NAME);
+    Map<String, String> properties = stageConfig.getProperties();
+    if (!Strings.isNullOrEmpty(properties.get(Properties.TimePartitionedFileSetDataset.BASE_PATH))) {
+      String tpfsName = properties.get(Properties.TimePartitionedFileSetDataset.TPFS_NAME);
       Preconditions.checkArgument(!Strings.isNullOrEmpty(tpfsName), "TimePartitionedFileSet name must be given.");
       pipelineConfigurer.createDataset(tpfsName, TimePartitionedFileSet.class.getName(), FileSetProperties.builder()
-        .setBasePath(stageConfig.getProperties().get(BASE_PATH))
+        .setBasePath(properties.get(Properties.TimePartitionedFileSetDataset.BASE_PATH))
```

Comment:
You don't have to use properties. You can have a private variable of type `TimePartitionedFileSetDatasetAvroConfig`. and you can access its fields.